ligo/stablecoin/actions.ligo

```diff
@@ -240,4 +338,237 @@ function is_operator_action
+  )
+
+(*
+ * Unpauses whole contract so that all transferring, burning, minting and
+ * allowance operations can be performed. Fails if the contract is already
+ * unpaused.
+ *)
+function unpause
+  ( const parameter : unpause_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ ensure_is_paused (store)
+; authorize_pauser (store)
+} with
+  ( (nil : list (operation))
+  , store with record [paused = False]
+  )
+
+(*
+ * Adds minter to minting allowances map setting its allowance
+ * to 0. Resets the allowance if associated address is already
+ * present. Fails if sender is not master minter.
+ *)
+function configure_minter
+  ( const parameter : configure_minter_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_master_minter (store)
+; const present_minting_allowance : option (nat) =
+    store.minting_allowances[parameter.0]
+; case parameter.1.0 of
+    None -> case present_minting_allowance of
+      Some (u) -> failwith ("NO_ALLOWANCE_EXPECTED")
+    | None -> skip
+    end
+  | Some (current_minting_allowance) ->
+      case present_minting_allowance of
+        Some (allowance) -> if allowance =/= current_minting_allowance
+          then failwith ("ALLOWANCE_MISMATCH")
+          else skip
+      | None -> failwith ("NOT_MINTER")
+      end
+  end
+} with
+  ( (nil : list (operation))
+  , store with record
+      [ minting_allowances = Big_map.update
+          ( parameter.0
+          , Some (parameter.1.1)
+          , store.minting_allowances
+          )
+      ]
+  )
+
+(*
+ * Removes minter from minting allowances map. Does nothing if
+ * minter is not present. Fails if sender is not master minter.
+ *)
+function remove_minter
+  ( const parameter : remove_minter_params
+  ; const store     : storage
+  ) : entrypoint is block
+{ authorize_master_minter (store)
+; case store.minting_allowances[parameter] of
+    Some (u) -> skip
+  | None -> failwith ("NOT_MINTER")
+  end
+} with
+  ( (nil : list (operation))
+  , store with record
+      [ minting_allowances = Big_map.remove
+          ( parameter
+          , store.minting_allowances
+          )
+      ]
+  )
+
+(*
+ * Produces tokens to a wallet associated with the given address
+ *)
+function mint
+  ( const parameters : mint_params
+  ; const store      : storage
+  ) : entrypoint is block
+{ ensure_not_paused (store)
+; authorize_minter (store)
+
+; const senderAddress : address = Tezos.sender
+
+; function mint_tokens
+    ( const accumulator       : storage
+    ; const mint_param        : mint_param
+    ; const current_allowance : nat
+    ) : storage is block
+  { const unwrapped_parameter : mint_param_ =
+      Layout.convert_from_right_comb ((mint_param : mint_param))
+  ; fail_on
+      ( unwrapped_parameter.amount > current_allowance
+      , "ALLOWANCE_EXCEEDED"
+      )
+
+  ; const updated_allowance : nat =
+      abs (current_allowance - unwrapped_parameter.amount)
```

Comment:
Might be better if we combine the `fail_on` call above above and this assignment using the `IS_NAT` check.

Reply:
done